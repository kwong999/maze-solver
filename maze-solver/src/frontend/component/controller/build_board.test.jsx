import React from 'react';
import renderer from 'react-test-renderer';
import BuildBoard from './build_board';

describe('BuildBoard', () => {
  const component = renderer.create(
    <BuildBoard
      dimensionRow={'9'}
      dimensionCol={'10'}
      handleChange={str => jest.fn()}
      handleBuildBoard={jest.fn()}
    />
  )
  test('should render without error', () => {
    let tree = component.toJSON();
    let componentInstance = component.root;
    let input = componentInstance.findAllByType('input');
    expect(input[0].props.value).toBe('9');
    expect(input[1].props.value).toBe('10');
    expect(tree).toMatchSnapshot();    
  })
})