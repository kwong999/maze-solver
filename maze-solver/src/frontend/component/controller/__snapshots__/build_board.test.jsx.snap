// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildBoard should render without error 1`] = `
<form
  className="build-board"
>
  <p>
    Dimension
  </p>
  <dl>
    <dt>
      Row:
    </dt>
    <dd>
      <input
        max="20"
        min="1"
        onChange={[MockFunction]}
        type="number"
        value="9"
      />
    </dd>
    <dt>
      Column:
    </dt>
    <dd>
      <input
        max="20"
        min="1"
        onChange={[MockFunction]}
        type="number"
        value="10"
      />
    </dd>
  </dl>
  <button
    onClick={[MockFunction]}
    type="submit"
  >
    Build Board
  </button>
</form>
`;
